js/usercss.js

```diff
@@ -0,0 +1,522 @@
+      }
+    }
+  }
+
+  function buildMeta(source) {
+    const style = _buildMeta(source);
+    validate(style);
+    return style;
+  }
+
+  function parseWord(state, error = 'invalid word') {
+    const match = state.text.slice(state.re.lastIndex).match(/^([\w-]+)\s*/);
+    if (!match) {
+      throw new Error(error);
+    }
+    state.value = match[1];
+    state.re.lastIndex += match[0].length;
+  }
+
+  function parseVar(state) {
+    const result = {
+      type: null,
+      label: null,
+      name: null,
+      value: null,
+      default: null,
+      options: null
+    };
+
+    parseWord(state, 'missing type');
+    result.type = state.type = state.value;
+    if (!META_VARS.includes(state.type)) {
+      throw new Error(`unknown type: ${state.type}`);
+    }
+
+    parseWord(state, 'missing name');
+    result.name = state.value;
+
+    parseString(state);
+    result.label = state.value;
+
+    if (state.type === 'checkbox') {
+      const match = state.text.slice(state.re.lastIndex).match(/([01])\s+/);
+      if (!match) {
+        throw new Error('value must be 0 or 1');
+      }
+      state.re.lastIndex += match[0].length;
+      result.default = match[1];
+    } else if (state.type === 'select' || (state.type === 'image' && state.key === 'var')) {
+      parseJSONValue(state);
+      if (Array.isArray(state.value)) {
+        result.options = state.value.map(text => createOption(text));
+      } else {
+        result.options = Object.keys(state.value).map(k =>
+          createOption(k, state.value[k])
+        );
+      }
+      result.default = result.options[0].name;
+    } else if (state.type === 'dropdown' || state.type === 'image') {
+      if (state.text[state.re.lastIndex] !== '{') {
+        throw new Error('no open {');
+      }
+      result.options = [];
+      state.re.lastIndex++;
+      while (state.text[state.re.lastIndex] !== '}') {
+        const option = {};
+
+        parseStringUnquoted(state);
+        option.name = state.value;
+
+        parseString(state);
+        option.label = state.value;
+
+        if (state.type === 'dropdown') {
+          parseEOT(state);
+        } else {
+          parseString(state);
+        }
+        option.value = state.value;
+
+        result.options.push(option);
+      }
+      state.re.lastIndex++;
+      eatWhitespace(state);
+      result.default = result.options[0].value;
+    } else {
+      // text, color
+      parseStringToEnd(state);
+      result.default = state.value;
+    }
+    state.usercssData.vars[result.name] = result;
+  }
+
+  function createOption(label, value) {
+    let name;
+    const match = label.match(/^(\w+):(.*)/);
+    if (match) {
+      ([, name, label] = match);
+    }
+    if (!name) {
+      name = label;
+    }
+    if (!value) {
+      value = name;
+    }
+    return {name, label, value};
+  }
+
+  function parseEOT(state) {
+    const match = state.text.slice(state.re.lastIndex).match(/^<<<EOT([\s\S]+?)EOT;/);
```

Comment:
To use lastIndex *properly* (without slicing the text) you need to specify `y` ("sticky") flag